src/text-editor-registry.js

```diff
@@ -214,14 +218,40 @@ class TextEditorRegistry {
     atom.grammars.autoAssignLanguageMode(editor.getBuffer())
   }
 
-  async subscribeToSettingsForEditorScope (editor) {
+  updateEditorSettingsForLanguageMode(editor, oldLanguageMode) {
+    const newLanguageMode = editor.buffer.getLanguageMode()
+
+    if (oldLanguageMode) {
+      const newSettings = this.textEditorParamsForScope(newLanguageMode.scopeDescriptor);
+      const oldSettings = this.textEditorParamsForScope(oldLanguageMode.scopeDescriptor);
+
+      let updatedSettings = {}
+      for (const [settingsKey, paramName, preserveSetting] of EDITOR_PARAMS_BY_SETTING_KEY) {
+        // Log the updated setting if it doesn't need to be preserved or if
+        // it has changed between the two language modes.  This prevents user-
+        // modified settings in an editor (like 'softWrapped') from being reset
+        // when the language mode changes.
+        if (!preserveSetting || !_.isEqual(newSettings[paramName], oldSettings[paramName])) {
```

Comment:
Can you explain why the `preserveSetting` flag is needed? I would think that the comparison between `oldSettings` and `newSettings` would be enough to fix the problem.

Reply:
Basically just trying to make it so that some settings always got overwritten if there's not a chance the user could tweak them.  However, if you think it's not necessary I'm more than happy to take it out!